cli/crates/common/src/environment.rs

```diff
@@ -212,16 +212,18 @@ impl Environment {
         let gql = get_path_to_file_in_project_grafbase(GRAFBASE_SCHEMA_FILE_NAME)?;
 
         if let Some(path) = ts_config {
-            let path = GrafbaseSchemaPath::ts_config(path);
+            match gql {
+                // Only warn, if the SDL and tsconfig are in the same directory.
+                Some(gql) if gql.parent() == path.parent() => {
+                    let warning = Warning::new("Found both grafbase.config.ts and schema.graphql files")
+                        .with_hint("Delete one of them to avoid conflicts");
 
-            if gql.is_some() {
-                let warning = Warning::new("Found both grafbase.config.ts and schema.graphql files")
-                    .with_hint("Delete one of them to avoid conflicts");
-
-                warnings.push(warning);
+                    warnings.push(warning);
+                }
+                _ => (),
             }
 
-            Ok(Some(path))
+            Ok(Some(GrafbaseSchemaPath::ts_config(path)))
```

Comment:
Does this not mean that if a user has

```
- grafbase
  - grafbase.config.ts
- other project
  - grafbase
    - schema.graphql
```

we'll pick the TS file even if they're clearly in other project?

Reply:
No. This utilizes the same mechanics to find the nearest `grafbase.config.ts` or `schema.graphql`. The bug happens when the closest one is the typescript config, and then somewhere else in the project you have a `schema.graphql`.

I'm rewriting this once more to make the directory scan only once.

Reply:
Really?  I may be missing something, but it looks to me like `get_path_to_file_in_project_grafbase` will go up the directory tree till it finds `grafbase/{file}`.  Which in the case above would find `grafbase/grafbase.config.ts` & `other project/grafbase/schema.graphql`.  And will then pick the TS file?

What am I missing?

Reply:
It will first find the `grafbase.config.ts` and if not found, do another search for `schema.graphql`. But... this changed a bit, check the new version: https://github.com/grafbase/grafbase/blob/fix/prevent-warning-if-tsconfig-and-schema-are-not-on-the-same-level/cli/crates/common/src/environment.rs#L211-L254